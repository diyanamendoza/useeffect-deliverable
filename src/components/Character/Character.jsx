export default function Character({ character }) {
    const { name, origin, species, gender, image, status } = character

    return (
        <figure aria-label="character">
          <img src={image} alt={`${name}-${species}`} />
          <div className="details">
            <h3>{name}</h3>
            <p>{species} | {gender}</p>
            <p>Origin: {origin}</p>
            <p>Status: {status}</p>
          </div>
        </figure>
      )
    }