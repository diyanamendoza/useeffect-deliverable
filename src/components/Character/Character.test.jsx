import { render } from '@testing-library/react';
import Character from './Character.jsx';

it('should render character list component', () => {
  const { container } = render(
    <Character character={{name: 'Ted', origin: 'Wisconsin', species: 'Rodent', gender: 'Unknown', image: 'https://hips.hearstapps.com/digitalspyuk.cdnds.net/17/20/1495207668-keanu-reeves-bill-bill-and-ted.jpg', status: 'Coasting'}} />
  );
  expect(container).toMatchSnapshot();
});