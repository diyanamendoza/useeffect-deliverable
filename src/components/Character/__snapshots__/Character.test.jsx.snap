// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render character list component 1`] = `
<div>
  <figure
    aria-label="character"
  >
    <img
      alt="Ted-Rodent"
      src="https://hips.hearstapps.com/digitalspyuk.cdnds.net/17/20/1495207668-keanu-reeves-bill-bill-and-ted.jpg"
    />
    <div
      class="details"
    >
      <h3>
        Ted
      </h3>
      <p>
        Rodent
         | 
        Unknown
      </p>
      <p>
        Origin: 
        Wisconsin
      </p>
      <p>
        Status: 
        Coasting
      </p>
    </div>
  </figure>
</div>
`;
